<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>MBP intro to statistics bootcamp</title>
    <meta charset="utf-8" />
    <meta name="author" content="Jason Lerch" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# MBP intro to statistics bootcamp
## <a href="https://jasonlerch.github.io/MBP-stats-2019/" class="uri">https://jasonlerch.github.io/MBP-stats-2019/</a>
### Jason Lerch
### Day 1

---


&lt;style&gt;
.small {
  font-size: 65%;
}

.medium {
  font-size: 80%;
}

.smallcode { 

}
.smallcode .remark-code {
  font-size: 50%
}

.footnote {
  font-size: 75%;
  color: gray;
}

&lt;/style&gt;



# Hello World

The three challenges of statistical inference are&lt;sup&gt;1&lt;/sup&gt;:

.footnote[
[1] From Andrew Gelman
]

--

1. Generalizing from sample to population

--

2. Generalizing from control to treatment group

--

3. Generalizing from observed measurements to underlying constructs of interest

---

# Three laws of statistics

.pull-left[

Arthur C. Clarke's three laws&lt;sup&gt;1&lt;/sup&gt;:

1.  When a distinguished but elderly scientist states that something is possible, he is almost certainly right. When he states that something is impossible, he is very probably wrong.
 
1. The only way of discovering the limits of the possible is to venture a little way past them into the impossible.

1. Any sufficiently advanced technology is indistinguishable from magic.
]

--

.pull-right[

Andrew Gelman's updates&lt;sup&gt;2&lt;/sup&gt;:

1. When a distinguished but elderly scientist states that “You have no choice but to accept that the major conclusions of these studies are true,” don’t believe him.

2. The only way of discovering the limits of the reasonable is to venture a little way past them into the unreasonable.

3. Any sufficiently crappy research is indistinguishable from fraud.
]

--

.small[
[1] https://en.wikipedia.org/wiki/Clarke%27s_three_laws

[2] http://andrewgelman.com/2016/06/20/clarkes-law-of-research/
]

---

# The MBP statistics bootcamp

Goals of this week:

1. Teach the theory and practice of statistics

1. Applied data analysis problem solving using R

1. Think hard about truth and replicability in science

Slides, recommended readings, and extra resources here:

https://jasonlerch.github.io/MBP-stats-2019/

(Will try and have slides for each day up the night before)

---

# The MBP statistics bootcamp

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 50%; margin-left: auto; margin-right: auto;  ">
<col><col><col><col><col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 1pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Hour</td>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Monday</td>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Tuesday</td>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Wednesday</td>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Thursday</td>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 10pt 4pt 10pt;">Friday</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 10pt 4pt 10pt;">9-12</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Introduction. Data organization, descriptive statistics, plotting, basic models. </td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Probability in all its glory. Multiple linear models, interactions, p values. </td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Hypothesis testing, searching for truth, multiple comparisons, and the crisis of replicability</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Putting it all together – analyzing a biomedical dataset from beginning to end. Review</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Presentations, exam</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 10pt 4pt 10pt;">12-3</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Group assignment #1</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Group assignment #2</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Group assignment #3</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;">Group assignment #4</td>
<td style="vertical-align: top; text-align: left; white-space: normal; padding: 4pt 10pt 4pt 10pt;"></td>
</tr>
</table>


---

# Grading

Exams (concepts only, no R):

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 50%; margin-left: auto; margin-right: auto;  ">
<col><col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">What</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">When</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">How much</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Short exam</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Tuesday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">5%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Short exam</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Wednesday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">5%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Short exam</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Thursday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">5%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Final exam</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Friday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">35%</td>
</tr>
</table>


Group assignments and presentations (R analyses and concepts):

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 50%; margin-left: auto; margin-right: auto;  ">
<col><col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">What</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">Due when</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 1pt 0pt; padding: 4pt 4pt 4pt 4pt;">How much</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Group assignment #1</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Tuesday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">10%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Group assignment #2</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Wednesday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">10%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Group assignment #3</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Thursday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">10%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Group assignment #4</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Friday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">10%</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 1pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Final presentation</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">Friday</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">10%</td>
</tr>
</table>


---

# Exams

* true/false, multiple choice, and short paragraphs.

* each class begins with ~ 10 minute, short exam covering previous day.

* final exam 30-60 minutes.

--

.pull-left[
Sample questions:



_Describe the null hypothesis_

_Identify elements of a box and whiskers plot (on a drawing)_

_Discuss analysis pre-registration advantages and disadvantages_

_TRUE/FALSE: if you compute a 95% confidence interval, you have a 95% chance of it containing the true value_
]

.pull-right[

![](Intro_files/figure-html/unnamed-chunk-4-1.png)&lt;!-- --&gt;


]


---

# Group assignments

* split into small groups of 3-4.

* we will assign groups.

* will try to mix groups by R and programming expertise.

* each group will be graded as a unit.

* final presentation given by a member of the group with least R/programming expertise.

---

class: inverse, center, middle

# Let's get started

---

# Statistical software

.pull-left[

Common software

1. Excel

1. SPSS

1. SAS

1. matlab

1. python

1. R
]

--

.pull-right[

Ups and downs of R

1. Open source, free, and powerful.

1. If a statistical test exists, it likely exists in R.

1. Literate programming/self documenting analyses.

1. Very strong in bioinformatics.

1. Steeper learning curve.
]



---

class: inverse, middle, center

# Reading and summarizing our data

---

# Intro to our dataset

.pull-left[

How do our brains change as we learn or undergo new experiences?

Earliest evidence that our brains are _plastic_ at larger, or _mesoscopic_, scales came from a study of taxi drivers in London, UK.

Mechanism of how that happens is unclear.

]

.pull-right[

&lt;img src="images/taxi.png" width="50%"&gt;

]

.footnote[PhD Thesis of Dulcie Vousden]

---

# Mouse models

.pull-left[

We can create taxi driving mice.

Use high-field MRI to get similar readout as in humans.

Use genetic models to test hypotheses of implicated pathways.

Use RNA sequencing to assess what changes per genotype or experimental group.

&lt;img src="images/taximouse.png"&gt;

]

.pull-right[

&lt;img src="images/mouseMRI.png" width="70%"&gt;

]

---

# The dataset



--

There are 283 mice in this dataset, with MRI scans acquired at 6 timepoints.

--

We have 3 genotypes: CREB -/-, CREB +/-, CREB +/+

--

There are 4 environmental conditions: Enriched, Exercise, Isolated Standard, Standard

--

MRIs were acquired at every timepoint, and the brains automatically segmented into  regions.

--

There are good reasons to believe that the hippocampus and the dentate gyrus of the hippocampus will be the most affected by the environmental interventions.

--

The effect of the three genotypes alone is interesting.


---

# Enrichment

&lt;video src="images/enrichment.mov" controls loop&gt;



---

# Reading data

A surprising amount of time in data analysis is spent in prepping data for visualization and analysis.


```r
library(tidyverse)
library(forcats)

mice &lt;- read_csv("mice.csv")
```

```
## Parsed with column specification:
## cols(
##   Age = col_double(),
##   Sex = col_character(),
##   Condition = col_character(),
##   Mouse.Genotyping = col_character(),
##   ID = col_double(),
##   Timepoint = col_character(),
##   Genotype = col_character(),
##   DaysOfEE = col_double(),
##   DaysOfEE0 = col_double()
## )
```

---
# Meet the mice


```r
str(mice, give.attr=FALSE)
```

```
## Classes 'spec_tbl_df', 'tbl_df', 'tbl' and 'data.frame':	1392 obs. of  9 variables:
##  $ Age             : num  8.5 8.5 8.5 9.5 9.5 8.5 8.5 9.5 8.5 9.5 ...
##  $ Sex             : chr  "M" "M" "M" "M" ...
##  $ Condition       : chr  "Enriched" "Standard" "Standard" "Enriched" ...
##  $ Mouse.Genotyping: chr  "Heterozygous" "Heterozygous" "Heterozygous" "Wildtype" ...
##  $ ID              : num  901 899 898 891 893 901 899 889 898 895 ...
##  $ Timepoint       : chr  "Pre1" "Pre1" "Pre1" "Pre1" ...
##  $ Genotype        : chr  "CREB +/-" "CREB +/-" "CREB +/-" "CREB +/+" ...
##  $ DaysOfEE        : num  -4 -4 -4 -4 -4 -3 -3 -3 -3 -3 ...
##  $ DaysOfEE0       : num  0 0 0 0 0 0 0 0 0 0 ...
```

---

# Numeric variable: age


```r
mice %&gt;%
  summarise(mean=mean(Age),
            min=min(Age),
            max=max(Age))
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 25.5555555555556%; margin-left: 0%; margin-right: auto;  ">
<col><col><col><tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">mean</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">min</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">max</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: normal; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">6.58</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">3.1</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">10.1</td>
</tr>
</table>



---

# Factors: Sex, Condition, Genotype

.pull-left[


```r
mice %&gt;%
  group_by(Sex) %&gt;%
  summarise(n=n())
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Sex</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">n</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">F</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">543</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">M</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">849</td>
</tr>
</table>

]


.pull-right[

```r
mice %&gt;% 
  group_by(Genotype) %&gt;% 
  summarize(n=n())
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Genotype</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">n</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">CREB -/-</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">426</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">CREB +/-</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">486</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">CREB +/+</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">480</td>
</tr>
</table>

]
---

# Subject descriptors: ID and Timepoint


```r
mice %&gt;%
  select(ID, Timepoint) %&gt;%
  head
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">ID</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Timepoint</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">899</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">898</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">891</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">893</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre2</td>
</tr>
</table>


---

# Alternate encodings: Genotype


```r
mice %&gt;%
  select(Genotype, Mouse.Genotyping) %&gt;%
  head
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 22.2222222222222%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Genotype</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Mouse.Genotyping</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">CREB +/-</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Heterozygous</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">CREB +/-</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Heterozygous</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">CREB +/-</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Heterozygous</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">CREB +/+</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Wildtype</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">CREB +/+</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Wildtype</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">CREB +/-</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">Heterozygous</td>
</tr>
</table>


---

# Alternate encodings: Days of EE, DaysofEE0


```r
mice %&gt;%
  filter(ID == 901) %&gt;%
  select(Timepoint, DaysOfEE, DaysOfEE0) %&gt;%
  head
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><col><tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Timepoint</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">DaysOfEE</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">DaysOfEE0</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">-4</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">0</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">Pre2</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">-3</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">0</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">24h</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">1</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">48h</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; padding: 4pt 4pt 4pt 4pt;">2</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">2</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">1 week</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">8</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">8</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">2 week</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">16</td>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">16</td>
</tr>
</table>


---

# Overview of subject numbers


```r
with(mice,
     ftable(Condition, Genotype, Timepoint))
```

```
##                            Timepoint 1 week 2 week 24h 48h Pre1 Pre2
## Condition         Genotype                                          
## Enriched          CREB -/-               24     25   7  24   24   24
##                   CREB +/-               30     33  12  34   30   33
##                   CREB +/+               27     30   8  30   27   28
## Exercise          CREB -/-               22     21   0  21   20   18
##                   CREB +/-               17     18   0  18   17   15
##                   CREB +/+               19     19   0  19   19   16
## Isolated Standard CREB -/-               14     14   0  14   13   14
##                   CREB +/-               12     12   0  12   11   12
##                   CREB +/+               17     17   0  17   17   14
## Standard          CREB -/-               23     26   4  26   25   23
##                   CREB +/-               29     34   9  34   32   32
##                   CREB +/+               28     31   6  31   31   29
```

---

# Factors, revisited

The Timepoint order makes no sense. Let's reorder


```r
mice &lt;- mice %&gt;%
  mutate(Timepoint=fct_relevel(Timepoint, "Pre1", "Pre2", "24h", 
                               "48h", "1 week", "2 week"))
with(mice, ftable(Condition, Genotype, Timepoint))
```

```
##                            Timepoint Pre1 Pre2 24h 48h 1 week 2 week
## Condition         Genotype                                          
## Enriched          CREB -/-             24   24   7  24     24     25
##                   CREB +/-             30   33  12  34     30     33
##                   CREB +/+             27   28   8  30     27     30
## Exercise          CREB -/-             20   18   0  21     22     21
##                   CREB +/-             17   15   0  18     17     18
##                   CREB +/+             19   16   0  19     19     19
## Isolated Standard CREB -/-             13   14   0  14     14     14
##                   CREB +/-             11   12   0  12     12     12
##                   CREB +/+             17   14   0  17     17     17
## Standard          CREB -/-             25   23   4  26     23     26
##                   CREB +/-             32   32   9  34     29     34
##                   CREB +/+             31   29   6  31     28     31
```

---

# Redo in tidyverse


```r
mice %&gt;% 
  group_by(Condition, Genotype, Timepoint) %&gt;% 
  summarise(n=n()) %&gt;% spread(Timepoint, value=n)
```

```
## # A tibble: 12 x 8
## # Groups:   Condition, Genotype [12]
##    Condition         Genotype  Pre1  Pre2 `24h` `48h` `1 week` `2 week`
##    &lt;chr&gt;             &lt;chr&gt;    &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;    &lt;int&gt;    &lt;int&gt;
##  1 Enriched          CREB -/-    24    24     7    24       24       25
##  2 Enriched          CREB +/-    30    33    12    34       30       33
##  3 Enriched          CREB +/+    27    28     8    30       27       30
##  4 Exercise          CREB -/-    20    18    NA    21       22       21
##  5 Exercise          CREB +/-    17    15    NA    18       17       18
##  6 Exercise          CREB +/+    19    16    NA    19       19       19
##  7 Isolated Standard CREB -/-    13    14    NA    14       14       14
##  8 Isolated Standard CREB +/-    11    12    NA    12       12       12
##  9 Isolated Standard CREB +/+    17    14    NA    17       17       17
## 10 Standard          CREB -/-    25    23     4    26       23       26
## 11 Standard          CREB +/-    32    32     9    34       29       34
## 12 Standard          CREB +/+    31    29     6    31       28       31
```


---

# Reading more data



```r
volumes &lt;- read_csv("volumes.csv")
```

```
## Parsed with column specification:
## cols(
##   .default = col_double(),
##   Timepoint = col_character()
## )
```

```
## See spec(...) for full column specifications.
```

---

# Inspecting the new data


```r
str(volumes)
```

```
## Classes 'spec_tbl_df', 'tbl_df', 'tbl' and 'data.frame':	1392 obs. of  161 variables:
##  $ amygdala                                          : num  9.84 10.3 10.53 10.39 10.57 ...
##  $ anterior commissure: pars anterior                : num  1.42 1.48 1.5 1.48 1.41 ...
##  $ anterior commissure: pars posterior               : num  0.392 0.428 0.382 0.444 0.463 ...
##  $ basal forebrain                                   : num  4.72 4.96 4.93 5.16 5.24 ...
##  $ bed nucleus of stria terminalis                   : num  1.24 1.31 1.28 1.35 1.32 ...
##  $ cerebellar peduncle: inferior                     : num  0.908 0.967 0.919 0.917 0.959 ...
##  $ cerebellar peduncle: middle                       : num  1.23 1.31 1.26 1.31 1.29 ...
##  $ cerebellar peduncle: superior                     : num  0.991 0.848 0.905 1.004 0.941 ...
##  $ cerebral aqueduct                                 : num  0.373 0.44 0.42 0.409 0.486 ...
##  $ cerebral peduncle                                 : num  2.58 2.54 2.6 2.57 2.55 ...
##  $ colliculus: inferior                              : num  5.46 5.6 5.34 5.66 5.29 ...
##  $ colliculus: superior                              : num  9.52 9.83 9.37 9.85 9.8 ...
##  $ corpus callosum                                   : num  14.1 14.3 14 14.6 14.1 ...
##  $ corticospinal tract/pyramids                      : num  1.59 1.59 1.56 1.6 1.6 ...
##  $ cuneate nucleus                                   : num  0.27 0.254 0.286 0.269 0.272 ...
##  $ dentate gyrus of hippocampus                      : num  3.96 3.92 3.95 4.03 3.88 ...
##  $ facial nerve (cranial nerve 7)                    : num  0.221 0.233 0.256 0.232 0.238 ...
##  $ fasciculus retroflexus                            : num  0.214 0.212 0.192 0.192 0.211 ...
##  $ fimbria                                           : num  2.67 2.9 2.73 2.74 2.85 ...
##  $ fornix                                            : num  0.639 0.625 0.686 0.646 0.633 ...
##  $ fourth ventricle                                  : num  0.879 0.832 0.87 0.878 0.892 ...
##  $ fundus of striatum                                : num  0.0999 0.148 0.129 0.14 0.1407 ...
##  $ globus pallidus                                   : num  3.19 3.2 3.04 3.15 3.07 ...
##  $ habenular commissure                              : num  0.0197 0.0306 0.0211 0.0175 0.0248 ...
##  $ hippocampus                                       : num  20.6 20.7 21.1 21.6 21.3 ...
##  $ hypothalamus                                      : num  10.2 10.8 10.5 10.8 10.6 ...
##  $ inferior olivary complex                          : num  0.303 0.356 0.33 0.375 0.271 ...
##  $ internal capsule                                  : num  3.04 3.04 2.91 3.08 3.12 ...
##  $ interpedunclar nucleus                            : num  0.278 0.275 0.297 0.299 0.27 ...
##  $ lateral olfactory tract                           : num  0.758 0.718 0.73 0.726 0.746 ...
##  $ lateral septum                                    : num  2.66 2.7 2.71 2.67 2.64 ...
##  $ lateral ventricle                                 : num  2.54 2.7 2.51 2.62 2.52 ...
##  $ mammillary bodies                                 : num  0.499 0.488 0.503 0.477 0.51 ...
##  $ mammilothalamic tract                             : num  0.219 0.231 0.215 0.22 0.218 ...
##  $ medial lemniscus/medial longitudinal fasciculus   : num  2.43 2.51 2.51 2.58 2.45 ...
##  $ medial septum                                     : num  1.02 1.09 1.06 1.03 1.02 ...
##  $ medulla                                           : num  28.1 28.9 28.9 28.2 28.3 ...
##  $ midbrain                                          : num  13.8 14.7 14.2 14.2 14.4 ...
##  $ nucleus accumbens                                 : num  4.06 4.01 4.23 3.99 3.97 ...
##  $ olfactory bulbs                                   : num  26.4 25.9 26.1 26.7 26.4 ...
##  $ olfactory tubercle                                : num  3.23 3.31 3.37 3.43 3.35 ...
##  $ optic tract                                       : num  1.8 1.88 1.76 1.86 1.86 ...
##  $ periaqueductal grey                               : num  4.05 4.13 3.95 4.19 4.33 ...
##  $ pons                                              : num  17.5 18.3 17.9 17.7 18.5 ...
##  $ pontine nucleus                                   : num  0.895 0.885 0.885 0.922 0.892 ...
##  $ posterior commissure                              : num  0.139 0.171 0.135 0.155 0.128 ...
##  $ pre-para subiculum                                : num  2.09 1.93 2 1.99 2.05 ...
##  $ stratum granulosum of hippocampus                 : num  0.903 0.883 0.928 0.914 0.899 ...
##  $ stria medullaris                                  : num  0.565 0.642 0.613 0.62 0.574 ...
##  $ stria terminalis                                  : num  1.07 1.06 1.11 1.12 1.14 ...
##  $ striatum                                          : num  22 22.3 22.2 22.8 22.8 ...
##  $ subependymale zone / rhinocele                    : num  0.0576 0.0561 0.0539 0.0642 0.0576 ...
##  $ superior olivary complex                          : num  0.909 0.899 0.98 0.857 0.878 ...
##  $ thalamus                                          : num  17.9 18.3 18.2 18.4 17.8 ...
##  $ third ventricle                                   : num  0.687 0.916 0.762 0.711 0.815 ...
##  $ ventral tegmental decussation                     : num  0.104 0.142 0.125 0.118 0.133 ...
##  $ lobules 1-2: lingula and central lobule (ventral) : num  2.34 2.23 2.21 2.32 2.38 ...
##  $ lobule 3: central lobule (dorsal)                 : num  2.41 2.7 2.59 2.34 2.45 ...
##  $ lobules 4-5: culmen (ventral and dorsal)          : num  5.14 5.22 5.22 5.1 4.97 ...
##  $ lobule 6: declive                                 : num  3.12 3.33 3.2 3.11 3.07 ...
##  $ lobule 7: tuber (or folium)                       : num  1.11 1.29 1.18 1.16 1.18 ...
##  $ lobule 8: pyramis                                 : num  1.7 1.86 1.74 1.7 1.68 ...
##  $ lobule 9: uvula                                   : num  3.21 3.09 3.12 2.87 2.93 ...
##  $ lobule 10: nodulus                                : num  1.63 1.7 1.79 1.64 1.66 ...
##  $ anterior lobule  (lobules 4-5)                    : num  1.96 1.89 1.97 1.77 1.98 ...
##  $ simple lobule (lobule 6)                          : num  5.84 6.42 5.73 6.06 5.79 ...
##  $ crus 1: ansiform lobule (lobule 6)                : num  5.16 5.58 5.27 5.14 5.3 ...
##  $ crus 2: ansiform lobule (lobule 7)                : num  3.44 3.57 3.68 3.56 3.81 ...
##  $ paramedian lobule (lobule 7)                      : num  4.39 4.55 4.69 4.57 4.6 ...
##  $ copula: pyramis (lobule 8)                        : num  2.97 3.06 3.13 2.85 2.88 ...
##  $ flocculus (FL)                                    : num  1.03 1.06 1.03 1.01 1.02 ...
##  $ paraflocculus (PFL)                               : num  5.04 4.77 4.92 5.02 5.08 ...
##  $ trunk of arbor vita                               : num  4.49 4.49 4.47 4.59 4.65 ...
##  $ lobule 1-2 white matter                           : num  0.1042 0.0962 0.1093 0.1072 0.1101 ...
##  $ lobule 3 white matter                             : num  0.286 0.284 0.291 0.252 0.268 ...
##  $ trunk of lobules 1-3 white matter                 : num  0.175 0.158 0.159 0.173 0.168 ...
##  $ lobules 4-5 white matter                          : num  0.796 0.785 0.803 0.851 0.803 ...
##  $ lobules 6-7 white matter                          : num  0.857 0.868 0.911 0.909 0.883 ...
##  $ lobule 8 white matter                             : num  0.137 0.125 0.114 0.128 0.119 ...
##  $ trunk of lobules 6-8 white matter                 : num  0.1392 0.1239 0.0933 0.1159 0.1239 ...
##  $ lobule 9 white matter                             : num  0.338 0.349 0.334 0.309 0.332 ...
##  $ lobule 10 white matter                            : num  0.0962 0.0955 0.1006 0.097 0.1057 ...
##  $ anterior lobule white matter                      : num  0.0977 0.0846 0.0911 0.0795 0.1028 ...
##  $ simple lobule white matter                        : num  0.505 0.574 0.517 0.517 0.5 ...
##  $ crus 1 white matter                               : num  0.509 0.505 0.486 0.504 0.507 ...
##  $ trunk of simple and crus 1 white matter           : num  0.181 0.176 0.187 0.194 0.192 ...
##  $ crus 2 white matter                               : num  0.249 0.251 0.25 0.265 0.27 ...
##  $ paramedian lobule                                 : num  0.112 0.128 0.124 0.114 0.122 ...
##  $ trunk of crus 2 and paramedian white matter       : num  0.356 0.355 0.389 0.398 0.385 ...
##  $ copula white matter                               : num  0.0904 0.0919 0.0802 0.0795 0.0729 ...
##  $ paraflocculus white matter                        : num  0.37 0.327 0.361 0.359 0.359 ...
##  $ flocculus white matter                            : num  0.0467 0.0488 0.043 0.0445 0.0467 ...
##  $ dentate nucleus                                   : num  0.275 0.289 0.278 0.308 0.283 ...
##  $ nucleus interpositus                              : num  0.441 0.456 0.432 0.476 0.478 ...
##  $ fastigial nucleus                                 : num  0.371 0.389 0.383 0.359 0.384 ...
##  $ Cingulate cortex: area 24a                        : num  1.76 1.83 1.67 1.74 1.64 ...
##  $ Cingulate cortex: area 24a'                       : num  0.856 0.846 0.816 0.819 0.74 ...
##  $ Cingulate cortex: area 24b                        : num  1.49 1.31 1.34 1.31 1.29 ...
##  $ Cingulate cortex: area 24b'                       : num  0.558 0.519 0.533 0.588 0.512 ...
##   [list output truncated]
##  - attr(*, "spec")=
##   .. cols(
##   ..   amygdala = col_double(),
##   ..   `anterior commissure: pars anterior` = col_double(),
##   ..   `anterior commissure: pars posterior` = col_double(),
##   ..   `basal forebrain` = col_double(),
##   ..   `bed nucleus of stria terminalis` = col_double(),
##   ..   `cerebellar peduncle: inferior` = col_double(),
##   ..   `cerebellar peduncle: middle` = col_double(),
##   ..   `cerebellar peduncle: superior` = col_double(),
##   ..   `cerebral aqueduct` = col_double(),
##   ..   `cerebral peduncle` = col_double(),
##   ..   `colliculus: inferior` = col_double(),
##   ..   `colliculus: superior` = col_double(),
##   ..   `corpus callosum` = col_double(),
##   ..   `corticospinal tract/pyramids` = col_double(),
##   ..   `cuneate nucleus` = col_double(),
##   ..   `dentate gyrus of hippocampus` = col_double(),
##   ..   `facial nerve (cranial nerve 7)` = col_double(),
##   ..   `fasciculus retroflexus` = col_double(),
##   ..   fimbria = col_double(),
##   ..   fornix = col_double(),
##   ..   `fourth ventricle` = col_double(),
##   ..   `fundus of striatum` = col_double(),
##   ..   `globus pallidus` = col_double(),
##   ..   `habenular commissure` = col_double(),
##   ..   hippocampus = col_double(),
##   ..   hypothalamus = col_double(),
##   ..   `inferior olivary complex` = col_double(),
##   ..   `internal capsule` = col_double(),
##   ..   `interpedunclar nucleus` = col_double(),
##   ..   `lateral olfactory tract` = col_double(),
##   ..   `lateral septum` = col_double(),
##   ..   `lateral ventricle` = col_double(),
##   ..   `mammillary bodies` = col_double(),
##   ..   `mammilothalamic tract` = col_double(),
##   ..   `medial lemniscus/medial longitudinal fasciculus` = col_double(),
##   ..   `medial septum` = col_double(),
##   ..   medulla = col_double(),
##   ..   midbrain = col_double(),
##   ..   `nucleus accumbens` = col_double(),
##   ..   `olfactory bulbs` = col_double(),
##   ..   `olfactory tubercle` = col_double(),
##   ..   `optic tract` = col_double(),
##   ..   `periaqueductal grey` = col_double(),
##   ..   pons = col_double(),
##   ..   `pontine nucleus` = col_double(),
##   ..   `posterior commissure` = col_double(),
##   ..   `pre-para subiculum` = col_double(),
##   ..   `stratum granulosum of hippocampus` = col_double(),
##   ..   `stria medullaris` = col_double(),
##   ..   `stria terminalis` = col_double(),
##   ..   striatum = col_double(),
##   ..   `subependymale zone / rhinocele` = col_double(),
##   ..   `superior olivary complex` = col_double(),
##   ..   thalamus = col_double(),
##   ..   `third ventricle` = col_double(),
##   ..   `ventral tegmental decussation` = col_double(),
##   ..   `lobules 1-2: lingula and central lobule (ventral)` = col_double(),
##   ..   `lobule 3: central lobule (dorsal)` = col_double(),
##   ..   `lobules 4-5: culmen (ventral and dorsal)` = col_double(),
##   ..   `lobule 6: declive` = col_double(),
##   ..   `lobule 7: tuber (or folium)` = col_double(),
##   ..   `lobule 8: pyramis` = col_double(),
##   ..   `lobule 9: uvula` = col_double(),
##   ..   `lobule 10: nodulus` = col_double(),
##   ..   `anterior lobule  (lobules 4-5)` = col_double(),
##   ..   `simple lobule (lobule 6)` = col_double(),
##   ..   `crus 1: ansiform lobule (lobule 6)` = col_double(),
##   ..   `crus 2: ansiform lobule (lobule 7)` = col_double(),
##   ..   `paramedian lobule (lobule 7)` = col_double(),
##   ..   `copula: pyramis (lobule 8)` = col_double(),
##   ..   `flocculus (FL)` = col_double(),
##   ..   `paraflocculus (PFL)` = col_double(),
##   ..   `trunk of arbor vita` = col_double(),
##   ..   `lobule 1-2 white matter` = col_double(),
##   ..   `lobule 3 white matter` = col_double(),
##   ..   `trunk of lobules 1-3 white matter` = col_double(),
##   ..   `lobules 4-5 white matter` = col_double(),
##   ..   `lobules 6-7 white matter` = col_double(),
##   ..   `lobule 8 white matter` = col_double(),
##   ..   `trunk of lobules 6-8 white matter` = col_double(),
##   ..   `lobule 9 white matter` = col_double(),
##   ..   `lobule 10 white matter` = col_double(),
##   ..   `anterior lobule white matter` = col_double(),
##   ..   `simple lobule white matter` = col_double(),
##   ..   `crus 1 white matter` = col_double(),
##   ..   `trunk of simple and crus 1 white matter` = col_double(),
##   ..   `crus 2 white matter` = col_double(),
##   ..   `paramedian lobule` = col_double(),
##   ..   `trunk of crus 2 and paramedian white matter` = col_double(),
##   ..   `copula white matter` = col_double(),
##   ..   `paraflocculus white matter` = col_double(),
##   ..   `flocculus white matter` = col_double(),
##   ..   `dentate nucleus` = col_double(),
##   ..   `nucleus interpositus` = col_double(),
##   ..   `fastigial nucleus` = col_double(),
##   ..   `Cingulate cortex: area 24a` = col_double(),
##   ..   `Cingulate cortex: area 24a'` = col_double(),
##   ..   `Cingulate cortex: area 24b` = col_double(),
##   ..   `Cingulate cortex: area 24b'` = col_double(),
##   ..   `Cingulate cortex: area 25` = col_double(),
##   ..   `Cingulate cortex: area 29a` = col_double(),
##   ..   `Cingulate cortex: area 29b` = col_double(),
##   ..   `Cingulate cortex: area 29c` = col_double(),
##   ..   `Cingulate cortex: area 30` = col_double(),
##   ..   `Cingulate cortex: area 32` = col_double(),
##   ..   `Amygdalopiriform transition area` = col_double(),
##   ..   `Primary auditory cortex` = col_double(),
##   ..   `Secondary auditory cortex: dorsal area` = col_double(),
##   ..   `Secondary auditory cortex: ventral area` = col_double(),
##   ..   `Caudomedial entorhinal cortex` = col_double(),
##   ..   Cingulum = col_double(),
##   ..   Claustrum = col_double(),
##   ..   `Cortex-amygdala transition zones` = col_double(),
##   ..   `Claustrum: dorsal part` = col_double(),
##   ..   `Dorsal nucleus of the endopiriform` = col_double(),
##   ..   `Dorsal intermediate entorhinal cortex` = col_double(),
##   ..   `Dorsolateral entorhinal cortex` = col_double(),
##   ..   `Dorsolateral orbital cortex` = col_double(),
##   ..   `Dorsal tenia tecta` = col_double(),
##   ..   `Ectorhinal cortex` = col_double(),
##   ..   `Frontal cortex: area 3` = col_double(),
##   ..   `Frontal association cortex` = col_double(),
##   ..   `Intermediate nucleus of the endopiriform claustrum` = col_double(),
##   ..   `Insular region: not subdivided` = col_double(),
##   ..   `Lateral orbital cortex` = col_double(),
##   ..   `Lateral parietal association cortex` = col_double(),
##   ..   `Primary motor cortex` = col_double(),
##   ..   `Secondary motor cortex` = col_double(),
##   ..   `Medial entorhinal cortex` = col_double(),
##   ..   `Medial orbital cortex` = col_double(),
##   ..   `Medial parietal association cortex` = col_double(),
##   ..   `Piriform cortex` = col_double(),
##   ..   `Posterolateral cortical amygdaloid area` = col_double(),
##   ..   `Posteromedial cortical amygdaloid area` = col_double(),
##   ..   `Perirhinal cortex` = col_double(),
##   ..   `Parietal cortex: posterior area: rostral part` = col_double(),
##   ..   `Rostral amygdalopiriform area` = col_double(),
##   ..   `Primary somatosensory cortex` = col_double(),
##   ..   `Primary somatosensory cortex: barrel field` = col_double(),
##   ..   `Primary somatosensory cortex: dysgranular zone` = col_double(),
##   ..   `Primary somatosensory cortex: forelimb region` = col_double(),
##   ..   `Primary somatosensory cortex: hindlimb region` = col_double(),
##   ..   `Primary somatosensory cortex: jaw region` = col_double(),
##   ..   `Primary somatosensory cortex: shoulder region` = col_double(),
##   ..   `Primary somatosensory cortex: trunk region` = col_double(),
##   ..   `Primary somatosensory cortex: upper lip region` = col_double(),
##   ..   `Secondary somatosensory cortex` = col_double(),
##   ..   `Temporal association area` = col_double(),
##   ..   `Primary visual cortex` = col_double(),
##   ..   `Primary visual cortex: binocular area` = col_double(),
##   ..   `Primary visual cortex: monocular area` = col_double(),
##   ..   `Secondary visual cortex: lateral area` = col_double(),
##   ..   `Secondary visual cortex: mediolateral area` = col_double(),
##   ..   `Secondary visual cortex: mediomedial area` = col_double(),
##   ..   `Claustrum: ventral part` = col_double(),
##   ..   `Ventral nucleus of the endopiriform claustrum` = col_double(),
##   ..   `Ventral intermediate entorhinal cortex` = col_double(),
##   ..   `Ventral orbital cortex` = col_double(),
##   ..   `Ventral tenia tecta` = col_double(),
##   ..   ID = col_double(),
##   ..   Timepoint = col_character()
##   .. )
```


---

# Linking data

.pull-left[


```r
volumes %&gt;%
  select(ID, Timepoint) %&gt;%
  head
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">ID</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Timepoint</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">899</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">898</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">891</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">893</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre2</td>
</tr>
</table>


]

.pull-right[


```r
mice %&gt;%
  select(ID, Timepoint) %&gt;%
  head
```

<table class="huxtable" style="border-collapse: collapse; margin-bottom: 2em; margin-top: 2em; width: 20%; margin-left: 0%; margin-right: auto;  ">
<col><col><tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">ID</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0.4pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt; font-weight: bold;">Timepoint</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">899</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">898</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">891</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0pt 0.4pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">893</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0pt 0pt; padding: 4pt 4pt 4pt 4pt; background-color: rgb(242, 242, 242);">Pre1</td>
</tr>
<tr>
<td style="vertical-align: top; text-align: right; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0pt 0.4pt 0.4pt; padding: 4pt 4pt 4pt 4pt;">901</td>
<td style="vertical-align: top; text-align: left; white-space: nowrap; border-style: solid solid solid solid; border-width: 0pt 0.4pt 0.4pt 0pt; padding: 4pt 4pt 4pt 4pt;">Pre2</td>
</tr>
</table>



]
---

# Joining data


```r
mice &lt;- mice %&gt;% 
  inner_join(volumes) 
```

```
## Joining, by = c("ID", "Timepoint")
```

```
## Warning: Column `Timepoint` joining factor and character vector, coercing
## into character vector
```

```r
str(mice)
```

```
## Classes 'spec_tbl_df', 'tbl_df', 'tbl' and 'data.frame':	1392 obs. of  168 variables:
##  $ Age                                               : num  8.5 8.5 8.5 9.5 9.5 8.5 8.5 9.5 8.5 9.5 ...
##  $ Sex                                               : chr  "M" "M" "M" "M" ...
##  $ Condition                                         : chr  "Enriched" "Standard" "Standard" "Enriched" ...
##  $ Mouse.Genotyping                                  : chr  "Heterozygous" "Heterozygous" "Heterozygous" "Wildtype" ...
##  $ ID                                                : num  901 899 898 891 893 901 899 889 898 895 ...
##  $ Timepoint                                         : chr  "Pre1" "Pre1" "Pre1" "Pre1" ...
##  $ Genotype                                          : chr  "CREB +/-" "CREB +/-" "CREB +/-" "CREB +/+" ...
##  $ DaysOfEE                                          : num  -4 -4 -4 -4 -4 -3 -3 -3 -3 -3 ...
##  $ DaysOfEE0                                         : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ amygdala                                          : num  9.84 10.3 10.53 10.39 10.57 ...
##  $ anterior commissure: pars anterior                : num  1.42 1.48 1.5 1.48 1.41 ...
##  $ anterior commissure: pars posterior               : num  0.392 0.428 0.382 0.444 0.463 ...
##  $ basal forebrain                                   : num  4.72 4.96 4.93 5.16 5.24 ...
##  $ bed nucleus of stria terminalis                   : num  1.24 1.31 1.28 1.35 1.32 ...
##  $ cerebellar peduncle: inferior                     : num  0.908 0.967 0.919 0.917 0.959 ...
##  $ cerebellar peduncle: middle                       : num  1.23 1.31 1.26 1.31 1.29 ...
##  $ cerebellar peduncle: superior                     : num  0.991 0.848 0.905 1.004 0.941 ...
##  $ cerebral aqueduct                                 : num  0.373 0.44 0.42 0.409 0.486 ...
##  $ cerebral peduncle                                 : num  2.58 2.54 2.6 2.57 2.55 ...
##  $ colliculus: inferior                              : num  5.46 5.6 5.34 5.66 5.29 ...
##  $ colliculus: superior                              : num  9.52 9.83 9.37 9.85 9.8 ...
##  $ corpus callosum                                   : num  14.1 14.3 14 14.6 14.1 ...
##  $ corticospinal tract/pyramids                      : num  1.59 1.59 1.56 1.6 1.6 ...
##  $ cuneate nucleus                                   : num  0.27 0.254 0.286 0.269 0.272 ...
##  $ dentate gyrus of hippocampus                      : num  3.96 3.92 3.95 4.03 3.88 ...
##  $ facial nerve (cranial nerve 7)                    : num  0.221 0.233 0.256 0.232 0.238 ...
##  $ fasciculus retroflexus                            : num  0.214 0.212 0.192 0.192 0.211 ...
##  $ fimbria                                           : num  2.67 2.9 2.73 2.74 2.85 ...
##  $ fornix                                            : num  0.639 0.625 0.686 0.646 0.633 ...
##  $ fourth ventricle                                  : num  0.879 0.832 0.87 0.878 0.892 ...
##  $ fundus of striatum                                : num  0.0999 0.148 0.129 0.14 0.1407 ...
##  $ globus pallidus                                   : num  3.19 3.2 3.04 3.15 3.07 ...
##  $ habenular commissure                              : num  0.0197 0.0306 0.0211 0.0175 0.0248 ...
##  $ hippocampus                                       : num  20.6 20.7 21.1 21.6 21.3 ...
##  $ hypothalamus                                      : num  10.2 10.8 10.5 10.8 10.6 ...
##  $ inferior olivary complex                          : num  0.303 0.356 0.33 0.375 0.271 ...
##  $ internal capsule                                  : num  3.04 3.04 2.91 3.08 3.12 ...
##  $ interpedunclar nucleus                            : num  0.278 0.275 0.297 0.299 0.27 ...
##  $ lateral olfactory tract                           : num  0.758 0.718 0.73 0.726 0.746 ...
##  $ lateral septum                                    : num  2.66 2.7 2.71 2.67 2.64 ...
##  $ lateral ventricle                                 : num  2.54 2.7 2.51 2.62 2.52 ...
##  $ mammillary bodies                                 : num  0.499 0.488 0.503 0.477 0.51 ...
##  $ mammilothalamic tract                             : num  0.219 0.231 0.215 0.22 0.218 ...
##  $ medial lemniscus/medial longitudinal fasciculus   : num  2.43 2.51 2.51 2.58 2.45 ...
##  $ medial septum                                     : num  1.02 1.09 1.06 1.03 1.02 ...
##  $ medulla                                           : num  28.1 28.9 28.9 28.2 28.3 ...
##  $ midbrain                                          : num  13.8 14.7 14.2 14.2 14.4 ...
##  $ nucleus accumbens                                 : num  4.06 4.01 4.23 3.99 3.97 ...
##  $ olfactory bulbs                                   : num  26.4 25.9 26.1 26.7 26.4 ...
##  $ olfactory tubercle                                : num  3.23 3.31 3.37 3.43 3.35 ...
##  $ optic tract                                       : num  1.8 1.88 1.76 1.86 1.86 ...
##  $ periaqueductal grey                               : num  4.05 4.13 3.95 4.19 4.33 ...
##  $ pons                                              : num  17.5 18.3 17.9 17.7 18.5 ...
##  $ pontine nucleus                                   : num  0.895 0.885 0.885 0.922 0.892 ...
##  $ posterior commissure                              : num  0.139 0.171 0.135 0.155 0.128 ...
##  $ pre-para subiculum                                : num  2.09 1.93 2 1.99 2.05 ...
##  $ stratum granulosum of hippocampus                 : num  0.903 0.883 0.928 0.914 0.899 ...
##  $ stria medullaris                                  : num  0.565 0.642 0.613 0.62 0.574 ...
##  $ stria terminalis                                  : num  1.07 1.06 1.11 1.12 1.14 ...
##  $ striatum                                          : num  22 22.3 22.2 22.8 22.8 ...
##  $ subependymale zone / rhinocele                    : num  0.0576 0.0561 0.0539 0.0642 0.0576 ...
##  $ superior olivary complex                          : num  0.909 0.899 0.98 0.857 0.878 ...
##  $ thalamus                                          : num  17.9 18.3 18.2 18.4 17.8 ...
##  $ third ventricle                                   : num  0.687 0.916 0.762 0.711 0.815 ...
##  $ ventral tegmental decussation                     : num  0.104 0.142 0.125 0.118 0.133 ...
##  $ lobules 1-2: lingula and central lobule (ventral) : num  2.34 2.23 2.21 2.32 2.38 ...
##  $ lobule 3: central lobule (dorsal)                 : num  2.41 2.7 2.59 2.34 2.45 ...
##  $ lobules 4-5: culmen (ventral and dorsal)          : num  5.14 5.22 5.22 5.1 4.97 ...
##  $ lobule 6: declive                                 : num  3.12 3.33 3.2 3.11 3.07 ...
##  $ lobule 7: tuber (or folium)                       : num  1.11 1.29 1.18 1.16 1.18 ...
##  $ lobule 8: pyramis                                 : num  1.7 1.86 1.74 1.7 1.68 ...
##  $ lobule 9: uvula                                   : num  3.21 3.09 3.12 2.87 2.93 ...
##  $ lobule 10: nodulus                                : num  1.63 1.7 1.79 1.64 1.66 ...
##  $ anterior lobule  (lobules 4-5)                    : num  1.96 1.89 1.97 1.77 1.98 ...
##  $ simple lobule (lobule 6)                          : num  5.84 6.42 5.73 6.06 5.79 ...
##  $ crus 1: ansiform lobule (lobule 6)                : num  5.16 5.58 5.27 5.14 5.3 ...
##  $ crus 2: ansiform lobule (lobule 7)                : num  3.44 3.57 3.68 3.56 3.81 ...
##  $ paramedian lobule (lobule 7)                      : num  4.39 4.55 4.69 4.57 4.6 ...
##  $ copula: pyramis (lobule 8)                        : num  2.97 3.06 3.13 2.85 2.88 ...
##  $ flocculus (FL)                                    : num  1.03 1.06 1.03 1.01 1.02 ...
##  $ paraflocculus (PFL)                               : num  5.04 4.77 4.92 5.02 5.08 ...
##  $ trunk of arbor vita                               : num  4.49 4.49 4.47 4.59 4.65 ...
##  $ lobule 1-2 white matter                           : num  0.1042 0.0962 0.1093 0.1072 0.1101 ...
##  $ lobule 3 white matter                             : num  0.286 0.284 0.291 0.252 0.268 ...
##  $ trunk of lobules 1-3 white matter                 : num  0.175 0.158 0.159 0.173 0.168 ...
##  $ lobules 4-5 white matter                          : num  0.796 0.785 0.803 0.851 0.803 ...
##  $ lobules 6-7 white matter                          : num  0.857 0.868 0.911 0.909 0.883 ...
##  $ lobule 8 white matter                             : num  0.137 0.125 0.114 0.128 0.119 ...
##  $ trunk of lobules 6-8 white matter                 : num  0.1392 0.1239 0.0933 0.1159 0.1239 ...
##  $ lobule 9 white matter                             : num  0.338 0.349 0.334 0.309 0.332 ...
##  $ lobule 10 white matter                            : num  0.0962 0.0955 0.1006 0.097 0.1057 ...
##  $ anterior lobule white matter                      : num  0.0977 0.0846 0.0911 0.0795 0.1028 ...
##  $ simple lobule white matter                        : num  0.505 0.574 0.517 0.517 0.5 ...
##  $ crus 1 white matter                               : num  0.509 0.505 0.486 0.504 0.507 ...
##  $ trunk of simple and crus 1 white matter           : num  0.181 0.176 0.187 0.194 0.192 ...
##  $ crus 2 white matter                               : num  0.249 0.251 0.25 0.265 0.27 ...
##  $ paramedian lobule                                 : num  0.112 0.128 0.124 0.114 0.122 ...
##  $ trunk of crus 2 and paramedian white matter       : num  0.356 0.355 0.389 0.398 0.385 ...
##  $ copula white matter                               : num  0.0904 0.0919 0.0802 0.0795 0.0729 ...
##   [list output truncated]
```


---

class: inverse, middle, center

# Data visualization

---

# Data visualization

Data visualization communicates your data to your audience - and can be how your data communicates with you.

--

Excellent guide to visualization:

https://www.data-to-viz.com

--

Your task for later will be to look at the interesting variables in this dataset. For now, we will look at sex and the brain instead.

---

# Histogram


```r
ggplot(mice) + 
  aes(x=`bed nucleus of stria terminalis`) + 
  geom_histogram() 
```

```
## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

![](Intro_files/figure-html/unnamed-chunk-22-1.png)&lt;!-- --&gt;

---

# Make it prettier


```r
ggplot(mice) + 
  aes(x=`bed nucleus of stria terminalis`) + 
  geom_histogram() +
* xlab(bquote(Volume ~ (mm^3))) +
* ggtitle("Bed nucleus of stria terminalis") +
* theme_gray(16)
```

```
## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

![](Intro_files/figure-html/unnamed-chunk-23-1.png)&lt;!-- --&gt;

---

# Histogram bins


```r
ggplot(mice) + 
  aes(x=`bed nucleus of stria terminalis`) + 
* geom_histogram(binwidth = 0.01) +
  xlab(bquote(Volume ~ (mm^3))) + 
  ggtitle("Bed nucleus of stria terminalis") +  
  theme_gray(16) 
```

![](Intro_files/figure-html/unnamed-chunk-24-1.png)&lt;!-- --&gt;

---

# Facets


```r
ggplot(mice) + 
  aes(x=`bed nucleus of stria terminalis`) + 
  geom_histogram(binwidth = 0.01) +
  xlab(bquote(Volume ~ (mm^3))) + 
  ggtitle("Bed nucleus of stria terminalis") +  
  theme_gray(16) +
* facet_grid(Sex ~ .)
```

![](Intro_files/figure-html/facets-1.png)&lt;!-- --&gt;

---

# Colours


```r
ggplot(mice) + 
* aes(x=`bed nucleus of stria terminalis`, fill=Sex) +
  geom_histogram(binwidth = 0.01) +
  xlab(bquote(Volume ~ (mm^3))) + 
  ggtitle("Bed nucleus of stria terminalis") +  
  theme_gray(16)
```

![](Intro_files/figure-html/colours-1.png)&lt;!-- --&gt;

---


![](Intro_files/figure-html/facets-1.png)

![](Intro_files/figure-html/colours-1.png)

---

# Points



```r
ggplot(mice) + 
  aes(x=Sex, y=`bed nucleus of stria terminalis`) +
  geom_point() + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") + 
  ylab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16) 
```

![](Intro_files/figure-html/unnamed-chunk-25-1.png)&lt;!-- --&gt;

???

* Point out changed aes
* Point out subtitle

---

# Points

That's not very useful - too many points to see separation.


```r
ggplot(mice) + 
  aes(x=Sex, y=`bed nucleus of stria terminalis`) +
* geom_jitter() +
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16)
```

![](Intro_files/figure-html/unnamed-chunk-26-1.png)&lt;!-- --&gt;

---

# Boxplot

Good view of data distribution


```r
ggplot(mice) + 
  aes(x=Sex, y=`bed nucleus of stria terminalis`) +
  geom_boxplot() + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16)
```

![](Intro_files/figure-html/unnamed-chunk-27-1.png)&lt;!-- --&gt;

---

![](images/boxplot.png)

---

# Ridge lines


```r
suppressMessages(library(ggridges))
ggplot(mice) + 
  aes(y=Sex, x=`bed nucleus of stria terminalis`) +
  geom_density_ridges() + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  xlab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16)
```

```
## Picking joint bandwidth of 0.0132
```

![](Intro_files/figure-html/unnamed-chunk-28-1.png)&lt;!-- --&gt;


---

# Violins


```r
ggplot(mice) + 
  aes(x=Sex, y=`bed nucleus of stria terminalis`) +
  geom_violin() + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16)
```

![](Intro_files/figure-html/unnamed-chunk-29-1.png)&lt;!-- --&gt;


---
class: smallcode

# Combining plot types

&lt;!-- Code is cut off in rendered slide --&gt;


.pull-left[

```r
ggplot(mice) + 
  aes(x=Sex, 
      y=`bed nucleus of stria terminalis`
      ) +
  geom_boxplot() +
  geom_jitter(width=0.2, 
              alpha=0.2) +  
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  ylab(bquote(
    Volume ~ (mm^3))) + 
  theme_classic(16)
```
]

.pull-right[

![](Intro_files/figure-html/c1-1.png)

]

---

# Adding colour


```r
ggplot(mice) + 
  aes(x=Sex, 
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
  geom_boxplot() +
  geom_jitter(width=0.2, 
              alpha=0.2) +  
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all timepoints and genotypes") +
  ylab(bquote(
    Volume ~ (mm^3))) + 
  theme_classic(16)
```

---

# Adding colour


![](Intro_files/figure-html/c2-1.png)

---

# Using colour for additional information


```r
ggplot(mice) + 
  aes(x=Sex, 
      y=`bed nucleus of stria terminalis`,
      colour=Timepoint) +
  geom_boxplot() +
  geom_jitter(alpha=0.2,
              position = position_jitterdodge(jitter.width = 0.2)) +  
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  theme_classic(16)
```

---

# Using colour for additional information

![](Intro_files/figure-html/c3-1.png)

---

# Using colour for additional information


```r
ggplot(mice) + 
  aes(x=Sex, 
      y=`bed nucleus of stria terminalis`,
      colour=Timepoint) +
  geom_boxplot() +
  geom_jitter(alpha=0.2,
              position = position_jitterdodge(jitter.width = 0.2)) +  
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  scale_colour_viridis_d(option="C", end=0.8) + 
  theme_classic(16)
```

---

# Using colour for additional information

![](Intro_files/figure-html/c4-1.png)

---

# Factor order, again

Apparently the factor ordering was lost in data joining?


```r
mice &lt;- mice %&gt;%
  mutate(Timepoint=fct_relevel(Timepoint, "Pre1", "Pre2", "24h", 
                               "48h", "1 week", "2 week"))

ggplot(mice) + 
  aes(x=Sex, 
      y=`bed nucleus of stria terminalis`,
      colour=Timepoint) +
  geom_boxplot() +
  geom_jitter(alpha=0.2,
              position = position_jitterdodge(jitter.width = 0.2)) +  
  ggtitle("Bed nucleus of stria terminalis",
          subtitle="Across all genotypes") +
  ylab(bquote(Volume ~ (mm^3))) + 
  scale_colour_viridis_d(option="C", end=0.8) +
  theme_classic(16)
```

---

# Factor ordering, again

![](Intro_files/figure-html/c5-1.png)

---

# Better encoding of time


```r
ggplot(mice) + 
  aes(x=DaysOfEE,
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
  geom_boxplot(aes(group=interaction(Timepoint, Sex))) + 
  geom_jitter(alpha=0.25, position = 
                position_jitterdodge(jitter.width = 0.2)) + 
  ylab(bquote(Volume ~ (mm^3))) + 
  xlab("Days of enrichment") + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle = "Change over time") + 
  theme_classic(16)
```


---

# Better encoding of time

![](Intro_files/figure-html/t1-1.png)

---

# Combining colours and facets


```r
ggplot(mice) + 
  aes(x=DaysOfEE,
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
  geom_boxplot(aes(group=interaction(Timepoint, Sex))) + 
  geom_jitter(alpha=0.25, position = 
                position_jitterdodge(jitter.width = 0.2)) + 
  ylab(bquote(Volume ~ (mm^3))) + 
  xlab("Days of enrichment") + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle = "Change over time") + 
* facet_grid(Genotype ~ .) +
  theme_classic(16)
```


---

# Combining colours and facets

![](Intro_files/figure-html/t2-1.png)

---

# Adding lines


```r
ggplot(mice) + 
  aes(x=DaysOfEE,
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
  geom_boxplot(aes(group=interaction(Timepoint, Sex))) + 
  geom_jitter(alpha=0.25, position = 
                position_jitterdodge(jitter.width = 0.2)) + 
* stat_summary(fun.y = median, geom="line",
*              position =
*                position_jitterdodge(jitter.width = 0.2)) +
  ylab(bquote(Volume ~ (mm^3))) + 
  xlab("Days of enrichment") + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle = "Change over time") + 
  facet_grid(Genotype ~ .) +
  theme_classic(16)
```

---

# Adding lines

![](Intro_files/figure-html/t3-1.png)

---

class: inverse, center, middle

# Descriptive statistics

---

# Descriptive statistics

summarize datapoints into measures of

* central tendency

  * mean/average: `\(\bar{x} = \frac{1}{n} \sum_{i=1}^n x_i\)`
  
  * median: sort numbers, pick middle
  
  * mode: most common number

* variance/dispersion

  * standard deviation: `\(\sigma = \sqrt{\frac{1}{n-1} \sum_{i=1}^n (x_i - \bar{x})^2}\)`
  
  * range: smallest to largest value
  
  * interquartile range: recursive median calculations (median of upper half, median of lower half)


---

# Plotting mean and standard deviation


```r
suppressMessages(library(Hmisc))

p &lt;- position_jitterdodge(jitter.width = 0.2)

ggplot(mice) + 
  aes(x=DaysOfEE,
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
* stat_summary(fun.data=mean_sdl, geom="pointrange", size=1,
*              shape=17, fun.args=(mult=1), position=p) +
  geom_jitter(alpha=0.15, position = p) + 
  stat_summary(fun.y = median, geom="line", 
               position =p) +
  ylab(bquote(Volume ~ (mm^3))) + 
  xlab("Days of enrichment") + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle = "Change over time") + 
  facet_grid(Genotype ~ .) +
  theme_classic(16)
```

```
## Warning: Removed 2 rows containing missing values (geom_pointrange).
```

---

# Plotting mean and standard deviation

![](Intro_files/figure-html/t4-1.png)

---

# Standard error of the mean

Standard deviations independent of sample size. Statistical tests usually take sample size into account. The standard error of the mean is the standard deviation divided by the square root of n.

`$$\sigma = \sqrt{\frac{1}{n-1} \sum_{i=1}^n (x_i - \bar{x})^2}$$`
`$$\sigma_{\bar{x}} = \frac{\sigma}{\sqrt{n}}$$`

---

# Plotting mean and standard error


```r
suppressMessages(library(Hmisc))

p &lt;- position_jitterdodge(jitter.width = 0.2)

ggplot(mice) + 
  aes(x=DaysOfEE,
      y=`bed nucleus of stria terminalis`,
      colour=Sex) +
* stat_summary(fun.data=mean_se, geom="pointrange",
*             fun.args=(mult=1), position=p) +
  geom_jitter(alpha=0.15, position = p) + 
  stat_summary(fun.y = median, geom="line", 
               position =p) +
  ylab(bquote(Volume ~ (mm^3))) + 
  xlab("Days of enrichment") + 
  ggtitle("Bed nucleus of stria terminalis",
          subtitle = "Change over time") + 
  facet_grid(Genotype ~ .) +
  theme_classic(16)
```

```
## Warning: Removed 2 rows containing missing values (geom_pointrange).
```

---

# Plotting mean and standard error

![](Intro_files/figure-html/t5-1.png)


---

# Summary table


```r
mice %&gt;%
  group_by(Genotype, Sex, DaysOfEE) %&gt;%
  summarise(m=mean(`bed nucleus of stria terminalis`)) %&gt;%
  spread(DaysOfEE, m)  %&gt;%
  knitr::kable(format = 'html')
```

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; Genotype &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; Sex &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; -4 &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; -3 &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; 1 &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; 2 &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; 8 &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; 16 &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB -/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.202775 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.217989 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.161297 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.231568 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.244746 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.262104 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB -/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.264994 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.283486 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.290038 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.316406 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.343368 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.339794 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.210892 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.244424 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.205037 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.256917 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.267488 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.280610 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.272896 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.301316 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.315517 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.318961 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.332175 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.332277 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/+ &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.223120 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.249269 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.222168 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.258373 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.277564 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.283006 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/+ &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.273769 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.313224 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.333948 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.330533 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.349106 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.344397 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---

# Summary table


```r
mice %&gt;%
  group_by(Genotype, Sex, DaysOfEE) %&gt;%
  summarise(m=paste(round(mean(`bed nucleus of stria terminalis`),2),
                    round(sd(`bed nucleus of stria terminalis`), 2), sep="±")) %&gt;%
  spread(DaysOfEE, m) %&gt;%
  knitr::kable(format = 'html')
```

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; Genotype &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; Sex &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; -4 &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; -3 &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; 1 &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; 2 &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; 8 &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; 16 &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB -/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.2±0.06 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.22±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.16±NA &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.23±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.24±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.26±0.05 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB -/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.26±0.06 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.28±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.29±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.32±0.06 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.34±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.34±0.05 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.21±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.24±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.21±NA &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.26±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.27±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.28±0.05 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/- &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.27±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.3±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.32±0.03 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.32±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.33±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.33±0.04 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/+ &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.22±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.25±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.22±0.03 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.26±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.28±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.28±0.05 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; CREB +/+ &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.27±0.06 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.31±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.33±0.02 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.33±0.04 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.35±0.05 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1.34±0.04 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---

class: inverse, center, middle

# Statistical tests

---

# From populations to samples

![](images/sample.png)

.footnote[
OpenIntro Statistics, Diez, Barr, and Cetinkaya-Rundel, 2015
]

???

Talk about sources of bias

---

# Data types

![](images/data-types.png)

Data types determine choice of statistics and/or encoding.

.footnote[
OpenIntro Statistics, Diez, Barr, and Cetinkaya-Rundel, 2015
]

---

# Sex ratios

Are the sex ratios in our data balanced?


```r
baseline &lt;- mice %&gt;% filter(Timepoint == "Pre1")
addmargins(with(baseline, table(Sex)))
```

```
## Sex
##   F   M Sum 
## 101 165 266
```

--

What should we expect?

Assume equal probability of male or female


```r
nrow(baseline) / 2
```

```
## [1] 133
```

---

# How likely was our real value?

Binomial distribution - flip of a coin.


```r
rbinom(1, 1, 0.5)
```

```
## [1] 1
```

```r
rbinom(1, 1, 0.5)
```

```
## [1] 0
```

```r
rbinom(1, 1, 0.5)
```

```
## [1] 1
```

```r
rbinom(10, 1, 0.5)
```

```
##  [1] 0 0 1 0 1 0 1 1 0 1
```


---

# How likely was our real value?


```r
baseline &lt;- mice %&gt;% filter(Timepoint == "Pre1")
addmargins(with(baseline, table(Sex)))
```

```
## Sex
##   F   M Sum 
## 101 165 266
```

Assuming random choice of male or female:


```r
distribution &lt;- rbinom(266, 1, 0.5)
sum(distribution==1)
```

```
## [1] 139
```

--


```r
rbinom(1, 266, 0.5)
```

```
## [1] 121
```

???

Get everyone in class to run it and get some answers

---

# Long run probability

We did a single experiment, and obtained 101 Females and 165 Males.

If we were to rerun the experiment again and again and again, and each experimental mouse had a 50/50 chance of being male or female, how often would we obtain 101 Females or fewer?

--


```r
nexperiments &lt;- 1000 
females &lt;- vector(length=nexperiments)
for (i in 1:nexperiments) {
  females[i] &lt;- rbinom(1, 266, 0.5)
}
head(females)
```

```
## [1] 132 138 135 136 125 140
```

--

Can be shortened as


```r
females2 &lt;- rbinom(nexperiments, 266, 0.5)
head(females2)
```

```
## [1] 125 129 149 139 131 133
```


---

# Long run probability


```r
head(females)
```

```
## [1] 132 138 135 136 125 140
```

```r
ggplot(data.frame(females=females)) +
  aes(x=females) + 
  geom_histogram(binwidth = 3) + 
  theme_minimal(16)
```

![](Intro_files/figure-html/unnamed-chunk-40-1.png)&lt;!-- --&gt;
---

# Long run probability


```r
ggplot(data.frame(females=females)) +
  aes(x=females) + 
  geom_histogram(binwidth = 3) + 
  theme_minimal(16)
```

![](Intro_files/figure-html/unnamed-chunk-41-1.png)&lt;!-- --&gt;

```r
sum(females&lt;=101)
```

```
## [1] 0
```

---

# Closed form solution

```r
ggplot() +
  geom_histogram(data=data.frame(females=females),
                 aes(x=females, y=..density..),
                 binwidth = 3) + 
  geom_bar(aes(c(100:160)), stat="function", 
*          fun=function(x) dbinom(round(x), 266, 0.5),
           alpha=0.5, fill="blue") +
  theme_minimal(16)
```

![](Intro_files/figure-html/unnamed-chunk-42-1.png)&lt;!-- --&gt;

---

# Closed form solution


```r
pbinom(101, 266, 0.5)
```

```
## [1] 5.223361e-05
```

--


```r
sum(dbinom(0:101, 266, 0.5))
```

```
## [1] 5.223361e-05
```

--

.pull-left[
![](Intro_files/figure-html/unnamed-chunk-45-1.png)&lt;!-- --&gt;

]

.pull-right[
![](Intro_files/figure-html/unnamed-chunk-46-1.png)&lt;!-- --&gt;
]

---

# Review

* We asked whether the sex ratio in the study was likely to be random, assuming an equal chance of an experimental mouse being male or female.

--

* We simulated 1000 studies under the assumption of n=266 and the odds of being female = 50%

--

* This is the null hypothesis.

--

* Our random data simulations test the null hypothesis: what would happen if we ran the experiment again and again and again under the same conditions assuming random assignment of males and females?

--

* Our p-value - the long run probability under repeated experiments - was vanishingly small.

--

So the choice of sex was almost certainly non-random. Does it matter?

---

# Contingency table


```r
baseline &lt;- mice %&gt;% filter(Timepoint == "Pre1")
with(baseline, table(Sex, Genotype))
```

```
##    Genotype
## Sex CREB -/- CREB +/- CREB +/+
##   F       29       31       41
##   M       53       59       53
```

```r
addmargins(with(baseline, table(Sex, Genotype)))
```

```
##      Genotype
## Sex   CREB -/- CREB +/- CREB +/+ Sum
##   F         29       31       41 101
##   M         53       59       53 165
##   Sum       82       90       94 266
```

---
# What would we expect?

The table of observed numbers


```r
addmargins(with(baseline, table(Sex, Genotype))) %&gt;% 
  knitr::kable(format = 'html')
```

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt;   &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; CREB -/- &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; CREB +/- &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; CREB +/+ &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Sum &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; F &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 29 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 31 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 101 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; M &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 53 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 59 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 53 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 165 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Sum &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 90 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 94 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 266 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

--

Calculating the expected numbers

|    | CREB -/- | CRE +/- | CREB +/+ | Sum |
|----|----------|---------|----------|-----|
|F   |82*101/266|90*101/266|94*101/266|101|
|M   |82*165/266|90*165/266|94*165/266|165|
|Sum |82        |90        |94        |266


---


# Using the chisq.test function for these calculations


```r
xtest &lt;- with(baseline, chisq.test(Sex, Genotype))
addmargins(xtest$observed)
```

```
##      Genotype
## Sex   CREB -/- CREB +/- CREB +/+ Sum
##   F         29       31       41 101
##   M         53       59       53 165
##   Sum       82       90       94 266
```

```r
addmargins(xtest$expected)
```

```
##      Genotype
## Sex   CREB -/- CREB +/- CREB +/+ Sum
##   F   31.13534 34.17293 35.69173 101
##   M   50.86466 55.82707 58.30827 165
##   Sum 82.00000 90.00000 94.00000 266
```

---

# `\(\chi^2\)` test

`$$\chi^2 = \sum^k_{i=1} \sum^l_{j=1} \frac{n_{ij} - \tilde{n}_{ij}}{\tilde{n}_{ij}} = \sum^k_{i=1} \sum^l_{j=1} \frac{(n_{ij} - \frac{n_i + n + j}{n})^2}{\frac{n_i + n + j}{n}}$$`

![](images/chisq.png)

--

```
sum( ((xtest$observed - xtest$expected)^2)/xtest$expected )
```

---

# `\(\chi^2\)` test


```r
sum( ((xtest$observed - xtest$expected)^2)/xtest$expected )
```

```
## [1] 1.983758
```

--

Put that number into context?

![](Intro_files/figure-html/unnamed-chunk-51-1.png)&lt;!-- --&gt;

---

# `\(\chi^2\)` test


```r
with(baseline, chisq.test(Sex, Genotype))
```

```
## 
## 	Pearson's Chi-squared test
## 
## data:  Sex and Genotype
## X-squared = 1.9838, df = 2, p-value = 0.3709
```


---

# Review

.medium[


* We asked whether the sex ratio in the study was likely to be random, assuming an equal chance of an experimental mouse being male or female.

* We simulated 1000 studies under the assumption of n=266 and the odds of being female = 50%

* This is the null hypothesis.

* Our random data simulations test the null hypothesis: what would happen if we ran the experiment again and again and again under the same conditions assuming random assignment of males and females?

* Our p-value - the long run probability under repeated experiments - was vanishingly small.

So the choice of sex was almost certainly non-random. Did it matter?

* Chi-squared test to assess contingency tables.

* test outcome against known distribution

* the sex bias was shared by all genotypes - p value indicated this could easily have occurred by random chance.
]

---

class: inverse, center, middle

# Literate programming

---

# Literate programming

.pull-left[

### The Idea:

* mix code, text, and figures in one document.

* All analyses and their outputs remain in sync

* Can work as a notebook

]

--

.pull-right[

### The Implementation:

* rmarkdown
    * simple markup language for text
    * code embedded in document
    * documents are compiled - or knitted - to produce output html or pdf
    
* Great alternative: Jupyter
]

---

class: inverse, center, middle

# Assignment

---

# Group assignment number 1

.medium[

1. Assemble into your assigned teams.

1. Ensure that RStudio is running and you can load all required libraries.

1. Load the required data

1. Create an rmarkdown document that contains the following:

    1. A summary table of the subject numbers per timepoint, genotype, and condition
  
    1. Visualization(s) of the difference in hippocampal volume by Genotype at the final timepoint.
  
    1. Visualization(s) of the difference in hippocampal volume by Condition at the final timepoint.
  
    1. Visualization(s) of the change over time by Condition and Genotype.
  
1. Make sure that all team members are listed as authors.

1. Any questions: ask here in person, or email us (jason.lerch@ndcn.ox.ac.uk, mehran.karimzadehreghbati@mail.utoronto.ca) and we promise to answer quickly.

]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="libs/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
